src/main/java/Duke.java

```diff
@@ -1,10 +1,377 @@
+    // inserts correct file path separator to data.txt file
+    private static final Path filePath = Paths.get(home, "Documents","ip","src", "main", "resources", "data.txt");
+    private static final Path dirPath = Paths.get(home, "Documents","ip","src", "main", "resources");
+    private static final boolean directoryExists = Files.exists(dirPath);
+
+    private static final String MESSAGE_LINEBREAK = "____________________________________________________________";
+    private static final String MESSAGE_GREETING = MESSAGE_LINEBREAK + "\n" +
+            "Hello! I'm Duke\n" +
+            "What can I do for you?\n" +
+            MESSAGE_LINEBREAK;
+    private static final String MESSAGE_BYE = "Bye. Hope to see you again soon!";
+    private static final String MESSAGE_CREATING_DIRECTORY = " not found. Creating directory...";
+    private static final String MESSAGE_CREATING_FILE = " not found. Creating file...";
+    private static final String MESSAGE_ERROR_INVALID_COMMAND = " is not a valid command";
+    private static final String MESSAGE_ERROR_DONE = "ERROR: 'done' must be followed by an integer";
+    private static final String MESSAGE_ERROR_TODO = "ERROR: 'todo' must be followed by a desciption";
+    private static final String MESSAGE_ERROR_DELETE = "ERROR: 'delete' must be followed by a desciption";
+    private static final String MESSAGE_ERROR_DEADLINE = "ERROR: 'deadline' must be followed by a desciption";
+    private static final String MESSAGE_ERROR_DEADLINE_MISSING_BY = "ERROR: 'deadline' must have a /by tag followed by a time";
+    private static final String MESSAGE_ERROR_EVENT = "ERROR: 'event' must be followed by a desciption";
+    private static final String MESSAGE_ERROR_EVENT_MISSING_AT = "ERROR: 'event' must have a /at tag followed by a time";
+    private static final String MESSAGE_ERROR_EMPTY_LIST = "ERROR: You have no tasks!";
+    private static final String MESSAGE_ERROR_FILE = "ERROR: data.txt not found";
+    private static final String TAG_BY = "/by ";
+    private static final String TAG_AT = "/at ";
+    private static ArrayList<Task> taskList = new ArrayList<Task>();
+    private static Scanner in = new Scanner(System.in);
+    private static String inputCommand = "";
+    private static int numberOfTasks = 0;
+
+    public static String getFirstWordOf(String input) {
+        if(input.contains(" ")) {
+            return inputCommand.substring(0,inputCommand.indexOf(' '));
+        }
+        else{
+            return input;
+        }
+    }
+
+    public static String removeFirstWordOf(String input) {
+        if(input.contains(" ")){
+            return input.substring(input.indexOf(' ') + 1);
+        }
+        else{
+            return "";
+        }
+    }
+
+    public static void printList() throws DukeException {
+        if(taskList.size() == 0){
+            throw new DukeException();
+        }
+
+        for(int i = 0; i < taskList.size(); i++) {
+            Task t = taskList.get(i);
```

Comment:
Maybe a specific name for the iterative item of Task rather than "t"? How about "taskItem"?